ositrace/ositrace_shacl.ttl

```diff
@@ -0,0 +1,286 @@
+            sh:maxCount 1 ;
+            sh:message "Validation of accuracyLaneModelHeight failed!"@en ;
+            sh:minCount 0 ;
+            sh:name "accuracyLaneModelHeight"@en ;
+            sh:order 2 ;
+            sh:path ositrace:accuracyLaneModelHeight ],
+        [ skos:example "0.01" ;
+            sh:datatype xsd:float ;
+            sh:description "Precision of measured road network (relative accuracy)"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of precision failed!"@en ;
+            sh:minCount 0 ;
+            sh:name "precision"@en ;
+            sh:order 0 ;
+            sh:path ositrace:precision ],
+        [ skos:example "0.1" ;
+            sh:datatype xsd:float ;
+            sh:description "Accuracy of lane modell 2d"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of accuracyLaneModel2d failed!"@en ;
+            sh:minCount 0 ;
+            sh:name "accuracyLaneModel2d"@en ;
+            sh:order 1 ;
+            sh:path ositrace:accuracyLaneModel2d ],
+        [ skos:example "Calibration performed at origin" ;
+            sh:datatype xsd:string ;
+            sh:description "Description of any calibration steps performed prior to measurement"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of calibration failed!"@en ;
+            sh:minCount 0 ;
+            sh:name "calibration"@en ;
+            sh:order 5 ;
+            sh:path ositrace:calibration ] ;
+    sh:targetClass ositrace:Quality .
+
+ositrace:QuantityShape a sh:NodeShape ;
+    sh:property [ skos:example "1000" ;
+            sh:datatype xsd:unsignedInt ;
+            sh:description "Number of frames/messages in the trace file"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of numberFrames failed!"@en ;
+            sh:name "numberFrames"@en ;
+            sh:order 2 ;
+            sh:path ositrace:numberFrames ] ;
+    sh:targetClass ositrace:Quantity .
+
+ositrace:MovingObjectShape a sh:NodeShape ;
+    sh:property [ skos:example "42" ;
+            sh:datatype xsd:unsignedLong ;
+            sh:description "Moving object identifier in trace file"@en ;
+            sh:mincount 1 ;
+            sh:maxcount 1 ;
+            sh:message "Validation of identifier failed!"@en ;
+            sh:name "identifier"@en ;
+            sh:order 1 ;
+            sh:path ositrace:identifier ],
+        [ skos:example "PMSF Mercedes CLS Ego Vehicle" ;
+            sh:datatype xsd:string ;
+            sh:description "Description of moving object in the trace file"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of description failed!"@en ;
+            sh:name "description"@en ;
+            sh:order 2 ;
+            sh:path ositrace:description ] ;
+    sh:targetClass ositrace:MovingObject .
+
+ositrace:EventShape a sh:NodeShape ;
+    sh:property [ skos:example "2024-05-11T13:20:03.250320Z" ;
+            sh:datatype xsd:dateTimeStamp ;
```

Comment:
Should we allow only real world time stamps? I think especially for simulation based trace files this will be difficult. Maybe also allow realive time stamps (in temporal doamin or even in the frame domain)?

Reply:
@heuerfin, what other kinds of timestamps do you have in mind? I would think, that having just the real world one, would make it easier putting events (start, stop and anthing in between) in relation and use that later on for the search in the catalogue.

Reply:
I thought about simulations detached from real world. In many cases at our side at least, timing is basically in simulation time `0` to `X` seconds. In that cases also OSI traces will be generated (thus probably `1970-01-01T00:00:00.000000Z` and its not clear if it is really at that time or virtual time). But I'm not sure if this is helpful. I thought that a frame number could be helpfull, since it can be based on a real world timing and thus be calculated to a real timestamp or a virtual timestamp with seconds. However, we loose then the time difference as you mentioned :/ Based on `startTime` and maybe a relative `event:time` we would achieve both.

However, I think the main question is: is it likely that a user will search for a absolute time based querry? I think the most important part is that the user will search for event strings and then checks if the timing is appropriate. With an absolute timing he has to check against `ositrace:startTime` and calculate the difference on his own. With relative timing he gets the answer directly. But this is only my assumption and I'm not sure what the most relavant queries will be.

Reply:
I would always try to keep meta-data definitions simple and well-defined, the ease of proccessing needs should not override the meta-data long-term storage needs. So absolute time stamps are a safe bet, and even for simulations you usually strive to have some real-world date time relationship, as otherwise all kinds of systems don't work properly.

One could try to add alternative ways of marking these things, however I would leave this for future evolution, because for actual tagging I don't think you will use GAIA-X meta-data fields, and if we want the equivalent of OpenLABEL or something else, this should be handled in another ontology, and referenced as additional data.

So for now I would leave it as the first definition, which can be refined going forward as needs arise.